go/logger/config.go

```diff
@@ -0,0 +1,222 @@
+package logger
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"os/user"
+
+	"github.com/spf13/viper"
+)
+
+// Supported auto config environment names
+const (
+	LogAutoCfgEnvName   string = "PRLOG_AUTOCFG"
+	KafkaAutoCfgEnvName        = "PRKAFKA_AUTOCFG"
+)
+
+// Supported auto configuration types
+const (
+	EnvConfig  string = "env"
+	FileConfig        = "file"
+	BothConfig        = "both"
+)
+
+// Supported environment name prefixes
+const (
+	LogEnvPrefix   string = "PRLOG"
+	KafkaEnvPrefix        = "PRKAFKA"
+)
+
+// Supported config file names
+const (
+	LogFileName   string = "pr_log_config"
+	KafkaFileName        = "pr_kafka_config"
+)
+
+var log Logger
+
+func defaultLogCfg() Configuration {
+	return Configuration{
+		LogPackage:           LogrusType,
+		LogLevel:             InfoType,
+		EnableCloudEvents:    true,
+		EnableKafka:          false,
+		KafkaFormat:          CEFormat,
+		EnableConsole:        true,
+		ConsoleFormat:        TextFormat,
+		ConsoleLevelTruncate: true,
+		EnableFile:           false,
+		FileFormat:           JSONFormat,
+		FileLevelTruncate:    false,
+		FileLocation:         "pavedroad.log",
+	}
+}
+
+func defaultKafkaCfg() ProducerConfiguration {
+	user, _ := user.Current()
+	return ProducerConfiguration{
+		Brokers:       []string{"localhost:9092"},
+		Topic:         "logs",
+		Partition:     RandomPartition,
+		Key:           FixedKey,
+		KeyName:       user.Username,
+		CloudeventsID: HMAC,
+		Compression:   CompressionSnappy,
+		AckWait:       WaitForLocal,
+		FlushFreq:     500, // milliseconds
+		EnableTLS:     false,
+	}
+}
+
+func init() {
+	var err error
+	config := new(Configuration)
+	err = EnvConfigure(defaultLogCfg(), config, os.Getenv(LogAutoCfgEnvName),
+		LogFileName, LogEnvPrefix)
+	// fmt.Printf("config %+v\n\n", config)
+	if err != nil {
+		fmt.Printf("Could not create logger configuration %s:", err.Error())
+		os.Exit(1)
+	}
+	kafkaConfig := new(ProducerConfiguration)
+	err = EnvConfigure(defaultKafkaCfg(), kafkaConfig, os.Getenv(KafkaAutoCfgEnvName),
+		KafkaFileName, KafkaEnvPrefix)
+	// fmt.Printf("config %+v\n\n", config)
+	if err != nil {
+		fmt.Printf("Could not create kafka configuration %s:", err.Error())
+		os.Exit(1)
+	}
+	config.KafkaProducerCfg = *kafkaConfig
+
+	if log, err = NewLogger(*config); err != nil {
+		fmt.Printf("Could not instantiate %s logger package: %s",
+			config.LogPackage, err.Error())
+		os.Exit(1)
+	}
+}
+
+func EnvConfigure(defaultCfg interface{}, config interface{}, auto string,
+	filename string, prefix string) error {
+
+	var defaultMap map[string]interface{}
+	defaultJson, err := json.Marshal(defaultCfg)
+	if err != nil {
+		return err
+	}
+	json.Unmarshal(defaultJson, &defaultMap)
```

Comment:
returns an error code that should be checked.

Reply:
Added error check